ssv/spectest/tests/partialsigcontainer/duplicatequorum.go

```diff
@@ -0,0 +1,36 @@
+package partialsigcontainer
+
+import (
+	"github.com/bloxapp/ssv-spec/qbft"
+	"github.com/bloxapp/ssv-spec/ssv/spectest/tests"
+	"github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv-spec/types/testingutils"
+)
+
+func DuplicateQuorum() tests.SpecTest {
+
+	// Create a test key set
+	ks := testingutils.Testing4SharesSet()
+
+	// Create PartialSignatureMessage for testing
+	msg1 := testingutils.PostConsensusAttestationMsg(ks.Shares[1], 1, qbft.FirstHeight)
+	msg12 := testingutils.PostConsensusAttestationMsg(ks.Shares[1], 1, qbft.FirstHeight)
+	msg2 := testingutils.PostConsensusAttestationMsg(ks.Shares[2], 2, qbft.FirstHeight)
+	msg3 := testingutils.PostConsensusAttestationMsg(ks.Shares[3], 3, qbft.FirstHeight)
+	msgs := []*types.PartialSignatureMessage{msg1.Message.Messages[0], msg12.Message.Messages[0], msg2.Message.Messages[0], msg3.Message.Messages[0]}
+
+	// Verify the reconstructed signature
+	expectedSig, err := types.ReconstructSignatures(map[types.OperatorID][]byte{1: msgs[0].PartialSignature, 2: msgs[2].PartialSignature, 3: msgs[3].PartialSignature})
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return &PartialSigContainerTest{
+		Name:            "duplicate quorum",
+		Quorum:          ks.Threshold,
+		ValidatorPubKey: ks.ValidatorPK.Serialize(),
+		SignatureMsgs:   msgs,
+		ExpectedResult:  expectedSig.Serialize(),
+		ExpectedQuorum:  true,
+	}
+}
```

Comment:
> The two test cases you said are not tested. It will raise an error when reconstructing the signature if we add the invalid one first.

@MatheusFranco99 should we add an issue for those two tests, or they exist in other PRs?

Reply:
@GalRogozinski 
Actually, I should have said the "single test case" he mentioned.

Receiving a first valid and then a second invalid message from the same author is not tested here but I don't see this as an issue. I think that the current `PartialSigContainer` module isn't supposed to do validation. On the other hand, the `Runner` is supposed to do it and we have the tests for it.